### PR TITLE
Unify terminal data into a single dict

### PR DESCRIPTION
**Pull request by lvxnull**
Also fixes issues caused by the prompt PR.

### CODE REVIEW

**Comment by Stunkymonkey** [comment]
I love it. This is what I wanted to do too.

I guess this is not compatible with python 3.8 right?

**Comment by lvxnull** [comment]
At the very least `cache` decorator is python 3.9.

**Comment by KyleGospo** [comment]
This may break tabs in prompt, `--new-window` and `--tab` must never be applied at the same time

**Comment by lvxnull** [comment]
@KyleGospo should be fixed now

File: nautilus_open_any_terminal/open_any_terminal_extension.py [hunk 1]
```diff
@@ -264,36 +149,40 @@ def set_terminal_args(*args):
...
+        new_tab_text += " (terminal does not support tabs)"
+    if flatpak != FLATPAK_PARMS[0] and terminal_data.flatpak_package is not None:
+        terminal_cmd = ["flatpak", "run", "--" + flatpak, terminal_data.flatpak_package]
+        flatpak_text = "with flatpak as {0}".format(flatpak)
+    else:
+        terminal_cmd = [terminal]
+        if terminal == "blackbox" and distro_id() == "fedora":
+            # It's called like this on fedora
+            terminal_cmd[0] = "blackbox-terminal"
+        flatpak = FLATPAK_PARMS[0]
```
Comment (Stunkymonkey) [inline nautilus_open_any_terminal/open_any_terminal_extension.py:179]: i think this line can be deleted?

File: nautilus_open_any_terminal/open_any_terminal_extension.py [hunk 2]
```diff
@@ -245,14 +131,13 @@ def distro_id():
 def open_terminal_in_file(filename):
     """open the new terminal with correct path"""
     cmd = terminal_cmd.copy()
-    if new_tab:
-        cmd.append(NEW_TAB_PARAMS[terminal])
-    else:
-        cmd.append(NEW_WINDOW_PARAMS[terminal])
+    if new_tab and terminal_data.new_tab_arguments:
+        cmd.extend(terminal_data.new_tab_arguments)
+    elif terminal_data.new_window_arguments:
```
Comment (Stunkymonkey) [inline nautilus_open_any_terminal/open_any_terminal_extension.py:136]: I guess this is the fix for #105 ?

**Review by Stunkymonkey — COMMENTED** [review]
Thanks you very much for your improvements.

File: nautilus_open_any_terminal/open_any_terminal_extension.py (hunk 2 above)
Comment (lvxnull) [reply nautilus_open_any_terminal/open_any_terminal_extension.py:136]: yes

File: nautilus_open_any_terminal/open_any_terminal_extension.py (hunk 1 above)
Comment (Stunkymonkey) [reply nautilus_open_any_terminal/open_any_terminal_extension.py:179]: nope. nevermind

**Comment by Stunkymonkey** [comment]
ready to merge?

**Comment by lvxnull** [comment]
ready

**Pull request merged**